File: src/main/java/com/windea/java/generator/TextGen.java
package com.windea.java.generator;

public interface TextGen<T> {
	String text();

	default int length() {
		return text().length();
	}

	default boolean isEmpty() {
		return text().isEmpty();
	}

	default boolean isBlank() {
		return text().isBlank();
	}
}

File: src/main/kotlin/com/windea/kotlin/generator/TextGen.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "MemberVisibilityCanBePrivate")

package com.windea.kotlin.generator

/**
 * 文本生成器接口。
 */
interface TextGen<T> {
	val text: String

	val length: Int
		get() = text.length

	val isEmpty: Boolean
		get() = text.isEmpty()

	val isBlank: Boolean
		get() = text.isBlank()
}
